File: backend/app/schemas/audit.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict # pyrefly: ignore [missing-import]
from app.models.audit_log import EntityType

class AuditLogBase(BaseModel):
    ticket_id: int | None
    actor_id: int
    actor_name: str
    actor_email: str
    action: str
    entity_type: EntityType
    entity_id: str
    old_value: dict | None = None
    new_value: dict | None = None
    event_metadata: dict | None = None

class AuditLogCreate(AuditLogBase):
    pass

class AuditLogRead(AuditLogBase):
    model_config = ConfigDict(from_attributes=True)
    id: int
    created_at: datetime

class AuditLogSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    ticket_id: int | None
    actor_name: str
    action: str
    entity_type: EntityType
    entity_id: str
    event_metadata: dict | None = None
    created_at: datetime

File: frontend/src/types/audit.ts
export type AuditAction =
  | "TICKET_CREATED"
  | "TICKET_UPDATED"
  | "TICKET_ASSIGNED"
  | "TICKET_CLAIMED"
  | "TICKET_REASSIGNED"
  | "TICKET_ESCALATED"
  | "STATUS_CHANGED"
  | "TICKET_RESOLVED"
  | "TICKET_CLOSED";

export type EntityType = "ticket" | "user" | "role";

export interface AuditLogRead {
  id: number;
  ticket_id: number | null;
  actor_id: number;
  actor_name: string;
  actor_email: string;
  action: AuditAction;
  entity_type: EntityType;
  entity_id: string;
  old_value: Record<string, any> | null;
  new_value: Record<string, any> | null;
  event_metadata: Record<string, any> | null;
  created_at: string;
}

export interface AuditLogSummary {
  id: number;
  ticket_id: number | null;
  actor_name: string;
  action: AuditAction;
  entity_type: EntityType;
  entity_id: string;
  event_metadata: Record<string, any> | null;
  created_at: string;
}
